src/contracts/impl/Car.java

```diff
@@ -0,0 +1,28 @@
+package Task_3.contracts.impl;
+
+import Task_3.contracts.IProduct;
+import Task_3.contracts.IProductPart;
+
+public class Car implements IProduct {
+    private IProductPart body;
+    private IProductPart chassis;
+    private IProductPart engine;
```

Comment:
Here your parts will be different only by name, because your CarLineStep creating generic CarPart, and in the end it's like 3 identical parts but with different labels on them :)

I suggest to create 3 implementation IProductPart  for each part and 3 implementation ILineStep  to create this different parts. You still can operate with them using abstractions - IProductPart   and ILineStep